tests_ce/unit_tests/test_exporter/test_csv_exporter.py

```diff
@@ -1,259 +1,360 @@
-#
-#     def test_no_fieldnames_provided(self):
-#         """Test exporting when fieldnames are not provided and need to be inferred."""
-#         data = [
-#             {"id": "1", "title": "Title 1", "year": 2020},
-#             {"id": "2", "title": "Title 2", "year": 2021},
-#         ]
-#         self.exporter.fieldnames = None  # Ensure fieldnames are not set
-#         product = ("test_product", data)
-#         self.exporter.consume(product)
-#         self.assertEqual(self.exporter.fieldnames, ["id", "title", "year"])  # Fieldnames inferred
-#         self.exporter.finalize_chunks()
-#
-#     def test_export_with_custom_quotechar(self):
-#         """Test exporting data with a custom quote character."""
-#         self.setup_context.properties = {"quotechar": "'", "quoting": csv.QUOTE_ALL}
-#         self.exporter = CSVExporter(
-#             setup_context=self.setup_context,
-#             product_name="test_product",
-#             chunk_size=1000,
-#             delimiter=None,
-#             quotechar=None,
-#             quoting=None,
-#             line_terminator=None,
-#             fieldnames=None,
-#             encoding=None,
-#         )
-#
-#         data = generate_mock_data(5)
-#         product = ("test_product", data)
-#         self.exporter.consume(product)
-#         self.exporter.finalize_chunks()
-#
-#     def test_export_with_different_quoting_options(self):
-#         """Test exporting data with different quoting options."""
-#         for quoting_option in [csv.QUOTE_MINIMAL, csv.QUOTE_NONNUMERIC, csv.QUOTE_NONE]:
-#             self.setUp(quoting=quoting_option)
-#
-#             data = generate_mock_data(5)
-#             product = ("test_product", data)
-#             self.exporter.consume(product)
-#             self.exporter.finalize_chunks()
-#
-#     def test_export_with_custom_encoding(self):
-#         """Test exporting data with a custom encoding."""
-#         self.setUp(encoding="utf-16")
-#
-#         data = generate_mock_data(10)
-#         product = ("test_product", data)
-#         self.exporter.consume(product)
-#         self.exporter.finalize_chunks()
-#
-#     def test_export_empty_data_list(self):
-#         """Test exporting when data list is empty."""
-#         product = ("test_product", [])
-#         self.exporter.consume(product)
-#         self.exporter.finalize_chunks()
-#
-#
-# if __name__ == "__main__":
-#     unittest.main()
+import csv
+import multiprocessing
+import os
+import tempfile
+import unittest
+import uuid
+from pathlib import Path
+
+from datamimic_ce.exporters.csv_exporter import CSVExporter
+from datamimic_ce.exporters.exporter_state_manager import ExporterStateManager
+
+
+def generate_mock_data(total_records=3000, title="Mock Title", year=2020):
+    """Generate mock data for testing."""
+    return [{"id": f"movie_{i + 1}", "title": f"{title} {i + 1}", "year": year} for i in range(total_records)]
+
+
+class MockSetupContext:
```

Comment:
something like this , should be reusable in a helper function ... because these default will be needed everywhere in the UT what do you think ?